src/ui/components/diary/TimelineEmotionItem.tsx

```diff
@@ -30,29 +30,32 @@ export function TimelineEmotionItem({ emotions }: TimelineEmotionItemProps) {
   }, [containerRef]);
 
   return (
-    <Container ref={containerRef}>
-      <>
-        {emotions &&
-          emotions.map((emotion, index) => (
-            <EmotionItem key={index} height={width}>
-              <img src={`/images/icon/emotion/${emotion}.svg`} alt={emotion} />
-            </EmotionItem>
-          ))}
-      </>
-    </Container>
+    emotions.length > 0 && (
+      <Container ref={containerRef}>
+        <>
+          {emotions &&
+            emotions.map((emotion, index) => (
+              <EmotionItem key={index} height={width}>
+                <img src={`/images/icon/emotion/${emotion}.svg`} alt={emotion} />
+              </EmotionItem>
+            ))}
```

Comment:
P3:

```suggestion
          {emotions &&
            emotions.map((emotion, index) => {
              const emotionImageSrc = `/images/icon/emotion/${emotion}.svg`;
              return (
                <EmotionItem key={index} height={width}>
                  <img src={emotionImageSrc} alt={emotion} />
                </EmotionItem>
              );
            })}
```

loop 중인 컴포넌트 로직에서도 변수로 분리해주시면 가독성에 도움 되실거예요!